libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/RouteLineComponent.kt

```diff
@@ -69,14 +72,21 @@ internal class RouteLineComponent(
         }
 
         coroutineScope.launch {
-            mapboxNavigation.flowRoutesUpdated().collect { result ->
-                val routeLines = result.navigationRoutes.map {
-                    NavigationRouteLine(it, null)
-                }
-                routeLineApi.setNavigationRouteLines(routeLines).let { routeDrawData ->
-                    ifNonNull(mapView.getMapboxMap().getStyle()) { style ->
-                        routeLineView.renderRouteDrawData(style, routeDrawData)
+            val routesFlow = mapboxNavigation.flowRoutesUpdated()
+                .map { it.navigationRoutes }
+                .stateIn(
+                    this,
+                    SharingStarted.WhileSubscribed(),
+                    mapboxNavigation.getNavigationRoutes()
+                )
+
+            routesFlow.collect { navigationRoutes ->
+                mapView.getMapboxMap().getStyle()?.also { style ->
+                    val routeLines = navigationRoutes.map {
+                        NavigationRouteLine(it, null)
                     }
+                    val routeDrawData = routeLineApi.setNavigationRouteLines(routeLines)
+                    routeLineView.renderRouteDrawData(style, routeDrawData)
```

Comment:
I thought we needed to use the clear functions 🤔 
https://github.com/mapbox/mapbox-navigation-android/blob/742d5a291c5891b69e7637fce8e8fc4e640ec7e2/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/api/MapboxRouteArrowApi.kt#L229-L236

``` kotlin
                routeArrowAPI.clearArrows().apply {
                    ifNonNull(routeArrowView, mapboxMap.getStyle()) { view, style ->
                        view.render(style, this)
                    }
                }
                routeLineAPI?.clearRouteLine()?.apply {
                    ifNonNull(routeLineView, mapboxMap.getStyle()) { view, style ->
                        view.renderClearRouteLineValue(style, this)
                    }
                }
```

Reply:
Oh that is route arrow. But it is also there for route line

https://github.com/mapbox/mapbox-navigation-android/blob/742d5a291c5891b69e7637fce8e8fc4e640ec7e2/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt#L476

Reply:
I learned from Seth that we don't need to invoke clear API on the route line. I was talking to him about the same API. Looks like if the route line api is given a list of empty routes, that is enough to clear them. Please correct me if I am wrong @cafesilencio

Reply:
Setting routes with an empty list and calling clear is the same end result but conceptually different in the context of the drop in UI.  It has been discussed in the past that the state of the route line should reflect `MapboxNavigation`. That means that route line mutations are driven by the routes observer.  The implementation of the route line component has reflected this concept.  The route line reacting to the routes observer has advantages like handling reroutes and `MapboxNavigation` remains the source of truth.

Another approach would be for some other entity to decide what the state of the route line should be and emit events that drive the route line appearance.  An argument can be made for this approach. In this case the route line would not react to the routes observer but instead this other entity.   This concept is more complicated in my opinion because there are other events that drive the route line's state like route progress and puck position updates.  

Using a hybrid of these two concepts I think would be the most complex.  I suggest choosing one or the other.